File: frontend/src/api/metrics.ts
import { apiRequest } from './client';
import type { DashboardMetrics } from '../types/metrics';

/**
 * Fetch aggregated dashboard metrics.
 *
 * Parameters:
 *   None.
 *
 * Returns:
 *   Dashboard metrics DTO.
 *
 * Edge cases:
 *   Backend returns zeros when external integrations are unavailable.
 */
export function getDashboardMetrics(): Promise<DashboardMetrics> {
  return apiRequest<DashboardMetrics>('/metrics');
}

File: backend/app/schemas/metrics.py
from pydantic import BaseModel


class DashboardMetrics(BaseModel):
    """Represent aggregated dashboard metrics.

    Parameters:
        tickets: Ticket counts (open, overdue, pending development, assigned to me).
        tasks: ClickUp task counts (pending, in progress, in sprint, blocked).
        time: Time tracking hours (today, week, month).
        actions: Pending assistant action counts.

    Returns:
        Dashboard metrics response.

    Edge cases:
        Some values may be zero when external integrations are unavailable.
    """

    tickets: dict[str, int]
    tasks: dict[str, int]
    time: dict[str, float]
    actions: dict[str, int]
